src/android_sdk/java/main/dev/perfetto/sdk/PerfettoTrackEventBuilder.java

```diff
@@ -525,14 +524,19 @@ public PerfettoTrackEventBuilder addField(long id, double val) {
 
   /** Adds a proto field with field id {@code id} and value {@code val}. */
   public PerfettoTrackEventBuilder addField(long id, String val) {
+    return addField(id, val, 0);
+  }
+
+  /** Adds a proto field with field id {@code id} and value {@code val}. */
+  public PerfettoTrackEventBuilder addField(long id, String val, long internedTypeId) {
```

Comment:
given interning has very subtle implications, can you please make it a dedicated method (addInternedField) rather than an overload which takes an extra integer.
the extra integer looks too "innocent". Unfortunately interning is not that innocent and has dramatic consequences on debugging